subprojects/platform-native/src/test/groovy/org/gradle/nativeplatform/toolchain/internal/NativeCompilerTest.groovy

```diff
@@ -41,7 +43,20 @@ public abstract class NativeCompilerTest extends Specification {
     protected abstract List<String> getCompilerSpecificArguments(File includeDir)
 
     protected CommandLineToolInvocationWorker commandLineTool = Mock(CommandLineToolInvocationWorker)
-    protected BuildOperationProcessor buildOperationProcessor = new DefaultBuildOperationProcessor(new DefaultBuildOperationQueueFactory(), new DefaultExecutorFactory(), 1)
+
+    private DefaultBuildOperationWorkerRegistry workerRegistry = new DefaultBuildOperationWorkerRegistry(1)
+    private BuildOperationWorkerRegistry.Completion outerOperationCompletion
+
+    protected BuildOperationProcessor buildOperationProcessor = new DefaultBuildOperationProcessor(new DefaultBuildOperationQueueFactory(workerRegistry), new DefaultExecutorFactory(), 1)
+
+    def "setup"() {
```

Comment:
I stubbed `BuildOperationWorkerRegistry` as an easy change to remove boilerplate from these tests. Mocking `BuildOperationProcessor` means also mocking `BuildOperationQueueFactory` and would add more boilerplate.